get_trace_data.py

```diff
@@ -0,0 +1,145 @@
+                print('{}\n##################\n\n'.format(trace_data))
+                continue
+            except (OSError, IOError) as e:
+                print('ERROR: bbox={} was marked as completed, but error while pulling data from disk: {}'.format(bbox,
+                                                                                                                  e))
+
+        trace_data, _ = get_trace_data_for_bbox(session, map_client_id, bbox)
+
+        pickle.dump(trace_data, open(os.path.join(output_dir, bbox + '.pickle'), 'wb'))
+
+        bbox_sections[bbox] = True
+        pickle.dump(bbox_sections, open(sections_filename, 'wb'))
+
+    return 'Done'
+
+
+def get_trace_data_for_bbox(session: requests.Session, map_client_id: str, bbox: str):
+    result, elapsed = {}, 0
+    # try:
+    start = time.time()
+
+    print('Getting seq for bbox={}'.format(bbox))
+    next_url = SEQUENCE_URL.format(map_client_id, bbox, SEQUENCES_PER_PAGE)
+    page = 1
+    while next_url:
+        print('Page {}, url={}'.format(page, next_url))
+        sequence_resp = session.get(next_url)
+        sequence_keys = []
+        for seq_f in sequence_resp.json()['features']:
+            # Skip sequences that have too few images
+            if len(seq_f['geometry']['coordinates']) >= SKIP_IF_FEWER_IMAGES_THAN:
+                sequence_keys.append(seq_f['properties']['key'])
+        images_resp = session.get(IMAGES_URL.format(map_client_id, ','.join(sequence_keys)))
+        for img_f in images_resp.json()['features']:
+            if img_f['properties']['sequence_key'] not in result:
+                result[img_f['properties']['sequence_key']] = []
+            result[img_f['properties']['sequence_key']].append((
+                parser.isoparse(img_f['properties']['captured_at']).timestamp(),
+                img_f['geometry']['coordinates']
+            ))
+
+        # Check if there is a next page or if we are finished with this bbox
+        next_url = sequence_resp.links['next']['url'] if 'next' in sequence_resp.links else None
+        page += 1
+
+    stop = time.time()
+    elapsed = stop - start
+    print('\n\n##################\nFinished processing seqs for bbox={}, elapsed time: {}'.format(bbox, elapsed))
+    print('{}\n##################\n\n'.format(result))
+    # with response_count.get_lock():
+    #     response_count.value += 1
+    # except Exception as e:
+    #     print(e)
+
+    return result, elapsed
+
+
+if __name__ == '__main__':
+    arg_parser = argparse.ArgumentParser()
+    # TODO: Make this optional and do the planet if so
+    arg_parser.add_argument('--bbox', type=str, help='Filter by the bounding box on the map, given as `min_longitude,'
+                                                     'min_latitude,max_longitude,max_latitude`', required=True)
+    arg_parser.add_argument('--mcid', type=str, help='Mapillary Client ID; obtained by registering an app at '
```

Comment:
can we refactor this to work as input to a factory pattern? my idea is like you pass a little json blob or something that tells what the provider is and what arguments it needs to do its job. something like:

`--traces-source='{"provider":"mapillary", "client_id":"somestring"}'`

The factory needs to only look at the provider part of the json, the rest of the json is free form and depends on what the provider needs. For now we'd have only one class or method that the factory would return mapping to mapillary. But then in the future when someone else wants to add a new provider they need only implement a factory method that can fetch data and produce the required tuple of time lat lon